### DISCUSSION TITLE
structView filed length

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#857] [Q&A] [answered]
The length of the field of the struct display widget cannot display the full RSA encryption result regardless of the large value set, the test program is

 https://github.com/ddkwork/crypt

 open the RSA page after running, there are default test parameters in it, click the encryption button

![{C4D59985-7CB6-4bce-A3D2-0697D139F676}.png](https://github.com/cogentcore/core/assets/19886504/45a10ee7-959d-48b8-952f-d48f241de3a5)

**Comment by ddkwork** [comment]
https://github.com/ddkwork/crypt/blob/master/main.go#L75-L81

**Comment by kkoreilly** [comment]
I will look into this, also in the context of #697.

**Reply by kkoreilly** [reply]
Have you tried adding a `grow:"1"` struct tag instead of the width one?

**Reply by ddkwork** [reply]
> Have you tried adding a grow:\"1\" struct tag instead of the width one?

i will try，thanks

https://github.com/ddkwork/crypt/blob/master/main.go#L80-L80

not working

**Reply by kkoreilly** [reply]
You also need to set the `StructView` itself to grow:

```go
view.Style(func(s *styles.Style) {
    s.Grow.Set(1, 0)
})
```

**Reply by ddkwork** [reply]
> You also need to set the `StructView` itself to grow:
> 
> ```go
> view.Style(func(s *styles.Style) {
>     s.Grow.Set(1, 0)
> })
> ```

Sorry for the late reply, I don't have time to play the computer today, and it still doesn't work after testing

https://github.com/ddkwork/crypt/blob/master/main.go#L93-L95

**Reply by kkoreilly** [reply, marked as answer]
I think I realized what the problem is: the max width is preventing the text field from growing enough. Again, I will work on improving the structure of this more in #697, but for now, can you try adding a `max-width:"0"` struct tag to disable the max width?

**Reply by ddkwork** [reply]
> I think I realized what the problem is: the max width is preventing the text field from growing enough. Again, I will work on improving the structure of this more in #697, but for now, can you try adding a `max-width:"0"` struct tag to disable the max width?

Working now, thanks!
